FloripaPublicTransportation/RouteDetailViewController.swift

```diff
@@ -0,0 +1,195 @@
+            self.saturdayDepartures = _departures!.filter { d in
+                return d.calendar == .Saturday
+            }
+            
+            self.sundayDepartures = _departures!.filter { d in
+                return d.calendar == .Sunday
+            }
+        }
+    }
+    
+    // MARK: tableView.reloadData() logic
+    
+    private var _finishedLoadingStops = false
+    private var _finishedLoadingDepartures = false
+    var finishedLoadingStops: Bool {
+        get { return _finishedLoadingStops }
+        set {
+            _finishedLoadingStops = newValue
+            if finishedLoadingStops && finishedLoadingDepartures {
+                dispatch_async(dispatch_get_main_queue(), {
+                    self.tableView.reloadData()
+                    self.activityIndicator.stopAnimating()
+                })
+            }
+        }
+    }
+    var finishedLoadingDepartures: Bool {
+        get { return _finishedLoadingDepartures }
+        set {
+            _finishedLoadingDepartures = newValue
+            if finishedLoadingStops && finishedLoadingDepartures {
+                dispatch_async(dispatch_get_main_queue(), {
+                    self.tableView.reloadData()
+                    self.activityIndicator.stopAnimating()
+                })
+            }
+        }
+    }
+    
+    // MARK: Private constants definitions
+    
+    private let reuseIdentifier = "detailCell"
+    private let headerTitles: [String] = [
+        NSLocalizedString("List of streets within the route", comment: "The section header title to show in RouteDetailViewController above the streets that compose the route"),
+            NSLocalizedString("Weekday timetable", comment: "The section header title to show in RouteDetailViewController above the weekday timetable"),
+            NSLocalizedString("Saturday timetable", comment: "The section header title to show in RouteDetailViewController above the Saturday timetable"),
+            NSLocalizedString("Sunday timetable", comment: "The section header title to show in RouteDetailViewController above the Sunday timetable")]
+
+    // MARK: - View delegate
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        guard Connectivity.isConnectedToNetwork() else {
+            Connectivity.popNoConnectionAlert(self)
+            return
+        }
+        
+        self.activityIndicator.hidesWhenStopped = true
+        self.view.addSubview(self.activityIndicator)
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        // at the moment viewDidLayoutSubviews() is called, the screen has already finished
+        // loading all its elements positions, so this is the perfect moment to (re)calculate
+        // the position of the activityIndicator (the center of the view):
+        
+        let activityIndicatorSize = self.activityIndicator.frame.width
+        self.activityIndicator.frame = CGRectMake(
```

Comment:
like, here, you are using frames. even though viewDidLayoutSubviews is called after the screen has really loaded and accommodated on the device screen is it safe to use frames? can you explain your reasoning behind it?